monkey/tests/unit_tests/monkey_island/cc/resources/test_agent_configuration.py

```diff
@@ -48,58 +40,37 @@ def test_agent_configuration_endpoint(flask_client):
         follow_redirects=True,
     )
     print(resp.data)
-    assert resp.status_code == 200
+    assert resp.status_code == HTTPStatus.NO_CONTENT
     resp = flask_client.get(AGENT_CONFIGURATION_URL)
 
-    assert resp.status_code == 200
+    assert resp.status_code == HTTPStatus.OK
 
     assert AgentConfiguration(**json.loads(resp.data)) == AgentConfiguration(**AGENT_CONFIGURATION)
 
 
-def test_agent_configuration_endpoint__bogus_data(flask_client, in_memory_agent_plugin_repository):
-    in_memory_agent_plugin_repository.save_plugin(FAKE_AGENT_PLUGIN_1)
-    agent_configuration = deepcopy(AGENT_CONFIGURATION)
-
-    agent_configuration["propagation"]["exploitation"]["exploiters"] = {
-        "bogus_exploiter": {"bogus_field": "bogus_value"}
-    }
-
-    resp = flask_client.put(
-        AGENT_CONFIGURATION_URL,
-        json=AgentConfiguration(**agent_configuration).dict(simplify=True),
-        follow_redirects=True,
-    )
-    assert resp.status_code == 400
-
-
-def test_agent_configuration_endpoint__invalid_plugin_data(
-    flask_client, in_memory_agent_plugin_repository
+@pytest.mark.parametrize(
+    "error, expected_status",
+    [
+        (Exception, HTTPStatus.INTERNAL_SERVER_ERROR),
+        (json.JSONDecodeError, HTTPStatus.BAD_REQUEST),
+        (PluginConfigurationValidationError, HTTPStatus.BAD_REQUEST),
+        (TypeError, HTTPStatus.BAD_REQUEST),
+        (ValueError, HTTPStatus.BAD_REQUEST),
+    ],
+)
+def test_agent_configuration_service_raises_exception(
+    build_flask_client, error: Type[Exception], expected_status: HTTPStatus
 ):
-    in_memory_agent_plugin_repository.save_plugin(FAKE_AGENT_PLUGIN_1)
-    agent_configuration = deepcopy(AGENT_CONFIGURATION)
-
-    agent_configuration["propagation"]["exploitation"]["exploiters"] = {
-        FAKE_AGENT_PLUGIN_1.plugin_manifest.name: {
-            "exploitation_success_rate": -123,
-            "propagation_success_rate": 321,
-        }
-    }
-
-    resp = flask_client.put(
-        AGENT_CONFIGURATION_URL,
-        json=AgentConfiguration(**agent_configuration).dict(simplify=True),
-        follow_redirects=True,
-    )
-    assert resp.status_code == 400
-
-
-def test_agent_configuration_invalid_config(flask_client):
-    resp = flask_client.put(AGENT_CONFIGURATION_URL, json={"invalid_config": "invalid_stuff"})
-
-    assert resp.status_code == 400
-
+    container = StubDIContainer()
+    mock_agent_configuration_service = MagicMock(spec=IAgentConfigurationService)
+    mock_agent_configuration_service.update_configuration = MagicMock(side_effect=error)
```

Comment:
According to the doc, update_configuration() only throws a `StorageError` or a `PluginConfigurationValidationError`. Does this need to be updated?

Reply:
Good call. Fixed.